dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -323,6 +366,75 @@ def _update_stop_condition(self, data: DataFrame):
             self._stopped_at_unique_ratio = merged_unique_ratio
             self._stopped_at_unique_count = merged_unique_count
 
+    @BaseColumnProfiler._timeit(name="categories")
+    def _get_categories_cms(self, df_series, len_df):
+        """Return count min sketch and heavy hitters for both the batch and stream case.
+
+        :param df_series: Series currently being processed by categorical profiler
+        :type df_series: Series
+        :return: cms, heavy_hitter_dict, missing_heavy_hitter_dict
+        """
+        cms = datasketches.count_min_sketch(self.num_hashes, self.num_buckets)
+        heavy_hitter_dict = defaultdict(int)
+        missing_heavy_hitter_dict = defaultdict(int)
+        for i, value in enumerate(df_series):
+            cms.update(value)
+            i_count = cms.get_estimate(value)
+            i_total_count = i_count + self.cms.get_estimate(value)
+            # approximate heavy-hitters
+            if i_count >= int(len_df / self._heavy_hitters_threshold):
+                heavy_hitter_dict[value] = i_count
+                missing_heavy_hitter_dict.pop(value, None)
+            elif i_total_count >= int(
+                (self.sample_size + len_df) / self._heavy_hitters_threshold
+            ):
+                missing_heavy_hitter_dict[value] = i_total_count
+                heavy_hitter_dict.pop(value, None)
+
+        return cms, heavy_hitter_dict, missing_heavy_hitter_dict
+
+    @BaseColumnProfiler._timeit(name="categories")
+    def _merge_categories_cms(
+        self,
+        cms1,
+        heavy_hitter_dict1,
+        len1,
+        missing_heavy_hitter_dict,
+        cms2,
+        heavy_hitter_dict2,
+        len2,
+    ):
+        """Return the aggregate count min sketch and approximate histogram (categories).
+
+        :param cms1: count min sketch
+        :type cms1: datasketches.countminsketch
+        :param cms2: count min sketch
+        :type cms2: datasketches.countminsketch
+        :param heavy_hitter_dict1: Heavy Hitters category count
+        :type heavy_hitter_dict1: Dict
+        :param heavy_hitter_dict2: Heavy Hitters category count
+        :type heavy_hitter_dict2: Dict
+        :param missing_heavy_hitter_dict2: Heavy Hitters category count
+        considering two batches are two chunks of a data stream
+        :type missing_heavy_hitter_dict2: Dict
+        :param len1: number of samples in batch 1
+        :type len1: int
+        :param len2: number of samples in batch 2
+        :type len2: int
+        :return: cms1, categories
+        """
+        cms2.merge(cms1)
+        categories = utils.add_nested_dictionaries(
+            heavy_hitter_dict2, heavy_hitter_dict1
+        )
+        categories.update(missing_heavy_hitter_dict)
```

Comment:
This actually might be problematic for the profile1 + profile2 case. We need to consider that categories of profile1 or profile2 may not have the full data bc it was `<` the limit. For cases where missing from one or the other, we might need to pull it from the `cms`